specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,175 @@
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value. | May be used as part of call to an output recording function only. |
+
+## Data Types and Values
+
+Within the base profile, defining local variables is not supported. Arguments to calls correspondingly are expected to be constant values. 
+
+Qubits and results are passes as a pointer of type `%Qubit*` and `%Result*` respectively, where the pointer itself rather than the memory location identifies the qubit or result: For the purpose of passing them as arguments in function calls, a 64-bit signed integer value is cast to the appropriate pointer type. The signed integer that is case must be in the interval `[0, nrQubits)` for `%Qubit*` and `[0, nrResults)` for `%Result*`, where `nrQubits` and `nrResults` are the required number of qubits and results defined by entry point attributes.
+
+Only the `%Qubit` and `%Result` data types are required to be supported by all backends. 
+
+Integers and double-precision floating point numbers are available as in full QIR;
+however, computations using these numeric types are not available.
+
+### Qubit and Result Usage
+
+runtime has autonomoie over how to represent qubits. also for base profile, the runtime is the only entity that needs to know how to interprete the opaque pointers (deref or not). that info is captured in the target triple/duo.
+
+Qubits may not be used after they have been measured. 
+Qubits and results need to be numbered consecutively, starting at 0.
+No qpu support needed for measuring individual qubits, but backend would support output selection?
+
+## Quantum Instruction Set
+
+For a Quantum Instruction Set to be compatible with the Base Profile, it needs to satisfy the following requirements:
+
+- Since the Base Profile doesn't permit to define local variables, all instructions are required to return void. 
+
+## Output Recording
+
+Log format is a separate spec. What the i8* can be is a separate spec. Default spec for front-ends to compile into.
+TODO: do string labels need to be zero terminated?
+
+The following functions are declared and used to record the program output: 
+| Function                  | Signature         | Description |
+|---------------------------|-------------------|-------------|
+| __quantum__rt__initialize_record_output    | `void(i8*)`  | Inserts a marker in the output log that contains an identifier for the used labeling scheme. The backend may choose which output format to use, and the label identifier is omitted for output formats that do not support labeling. |
+| __quantum__rt__tuple_record_output    | `void(i64, i8*)`  | Inserts a marker in the output log that indicates the start of a tuple and how many tuple elements are going to be logged. The second parameter reflects an label for the tuple. The backend may choose which output format to use. Depending on the used format, the label will be logged or omitted. |
```

Comment:
the tuple recording and array recording that use an integer to say the length of the record only makes any sense for ordered output. For tagged/labelled output, it doesn't make sense to include these at all, so I'm not sure what function they provide. In the example above, it seems to mix labelled and unlabelled output in a way that doesn't work for execution. Should the example (and the documentation in the table) differentiate between these to avoid confusion from mixing them?

Reply:
I missed the `TODO` below that tries to clarify intent here. If the goal is that the IR has enough information for every style of output, then I think this table should clarify that the markers (in addition to the labels) might be omitted entirely (ie: some of these runtime functions may be a no-op for certain output styles).

Reply:
To decouple of the IR specification from the output format, I believe that we would indeed expect that the markers are always present in the IR (just like the labels) would always be there, and the backend would choose how to leverage that information to produce the output format of its choice.

Reply:
I revised this paragraph and hope it is clear now. I haven't fleshed out the spec for the output schemas, but will give that a go in the near future.